v3/src/components/graph/adornments/univariate-measures/box-plot/box-plot-adornment-model.ts

```diff
@@ -0,0 +1,115 @@
+import { Instance, types } from "mobx-state-tree"
+import { median, quantileSeq } from "mathjs"
+import { UnivariateMeasureAdornmentModel } from "../univariate-measure-adornment-model"
+import { kBoxPlotType, kBoxPlotValueTitleKey } from "./box-plot-adornment-types"
+import { IDataConfigurationModel } from "../../../../data-display/models/data-configuration-model"
+import { IAdornmentModel } from "../../adornment-models"
+
+export const BoxPlotAdornmentModel = UnivariateMeasureAdornmentModel
+  .named("BoxPlotAdornmentModel")
+  .props({
+    type: types.optional(types.literal(kBoxPlotType), kBoxPlotType),
+    labelTitle: types.optional(types.literal(kBoxPlotValueTitleKey), kBoxPlotValueTitleKey),
+    showOutliers: types.optional(types.boolean, false)
+  })
+  .views(() => ({
+    get hasRange() {
+      return true
+    },
+    getQuantileValue(quantile: number, caseValues: number[]) {
+      const sortedCaseValues = caseValues.sort((a, b) => a - b)
+      const lastIndex = sortedCaseValues.length - 1
+      const i = lastIndex * quantile / 100
+      const i1 = Math.floor(i)
+      const i2 = Math.ceil(i)
+      const fraction = i - i1
+
+      if (i < 0) {
+        return 0 // length === 0, or quantile < 0.0
+      } else if (i >= lastIndex) {
+        return sortedCaseValues[lastIndex] // quantile >= 1.0
+      } else if (i === i1) {
+        return sortedCaseValues[i1] // quantile falls on data value exactly
+      } else {
+        // quantile between two data values;
+        // note that quantile algorithms vary on method used to get value here, there is no fixed standard.
+        return (sortedCaseValues[i2] * fraction + sortedCaseValues[i1] * (1.0 - fraction))
```

Comment:
Yeah 👍 You probably don't recognize them because they were added so long ago, but I lifted these lines from V2.

See https://github.com/concord-consortium/codap/blob/8498540b0fe401fe914c70c3d4a70f66e6eb62fb/apps/dg/utilities/math_utilities.js#L226-L228